packages/labs/analyzer/src/lib/javascript/classes.ts

```diff
@@ -302,34 +306,81 @@ export const getHeritage = (
 
 export const getHeritageFromExpression = (
   expression: ts.Expression,
+  isMixinClass: boolean,
   analyzer: AnalyzerInterface
 ): ClassHeritage => {
   // TODO(kschaaf): Support for extracting mixing applications from the heritage
   // expression https://github.com/lit/lit/issues/2998
   const mixins: Reference[] = [];
-  const superClass = getSuperClass(expression, analyzer);
+  const superClass = getSuperClassAndMixins(expression, mixins, analyzer);
   return {
-    superClass,
+    superClass: isMixinClass ? undefined : superClass,
     mixins,
   };
 };
 
-export const getSuperClass = (
+export const getSuperClassAndMixins = (
   expression: ts.Expression,
+  foundMixins: Reference[],
   analyzer: AnalyzerInterface
 ): Reference | undefined => {
   // TODO(kschaaf) Could add support for inline class expressions here as well
   if (analyzer.typescript.isIdentifier(expression)) {
     return getReferenceForIdentifier(expression, analyzer);
+  } else if (
+    analyzer.typescript.isCallExpression(expression) &&
+    analyzer.typescript.isIdentifier(expression.expression)
```

Comment:
to what degree do we have to worry about weird cases like inline mixins?
```ts
class A {}
export class C extends (s => class extends s {})(A) {}
```

Reply:
one for @kevinpschaaf i think - imo we shouldn't worry about it since there's all sorts of crazy syntax you could put there so it may be a very deep rabbit hole